code/game/machinery/doors/firedoor.dm

```diff
@@ -531,23 +531,19 @@
 				constructionStep = CONSTRUCTION_WIRES_EXPOSED
 				update_icon()
 				return
-			if(C.tool_behaviour == TOOL_WRENCH)
-				var/obj/machinery/door/firedoor/A = locate(/obj/machinery/door/firedoor) in get_turf(src)
-				if(A && A.dir == src.dir)
-					to_chat(user, "<span class='warning'>There's already a firelock there.</span>")
+			if(C.tool_behaviour == TOOL_WRENCH) //when in doubt, copypaste from modern tgcode
+				if(locate(/obj/machinery/door/firedoor) in get_turf(src))
```

Comment:
the issue here is that, since we have directional firelocks, you SHOULD be able to place multiple firelocks on a tile. i think. same with below

Reply:
just tested it, even with the original code you cant seem to place directional firelocks on the same tile, even facing different directions.

Reply:
upon further testing and inspection, firelock code is so, so much worse than I thought. Either you can make fulltile firelocks, or directional firelocks and there is no in between. Reinserting some code I deleted makes fulltile firelocks work, but turns directional firelocks into fulltile ones and returning it back to normal only allows directional firelocks to be finished. Curse you whitesands!!!

Reply:
Your weakness is shown, for you cannot handle the majestic fuckyou firelocks :joker:

Reply:
listen, im looking at the gitblame for these and seeing years and years of people layering changes on this and wondering why I even bother. This rosetta stone of shitcode is something beyond me, beyond God.